backend/public-graph/graph/resolver.go

```diff
@@ -295,37 +296,38 @@ func (r *Resolver) AppendProperties(ctx context.Context, sessionID int, properti
 	return nil
 }
 
-func (r *Resolver) CreateSessionEvents(ctx context.Context, sessionID int, events []*clickhouse.SessionEventRow) error {
-	outerSpan, ctxT := util.StartSpanFromContext(ctx, "public-graph.CreateSessionEvents", util.Tag("sessionID", sessionID))
+func (r *Resolver) SubmitSessionEvents(ctx context.Context, sessionID int, events []*clickhouse.SessionEventRow) error {
+	outerSpan, ctxT := util.StartSpanFromContext(ctx, "public-graph.SubmitSessionEvents", util.Tag("sessionID", sessionID))
 	defer outerSpan.Finish()
 
-	loadSessionSpan, ctxS := util.StartSpanFromContext(ctxT, "public-graph.CreateSessionEvents.loadSessions", util.Tag("sessionID", sessionID))
+	loadSessionSpan, ctxS := util.StartSpanFromContext(ctxT, "public-graph.SubmitSessionEvents.loadSessions", util.Tag("sessionID", sessionID))
 	session := &model.Session{}
 	res := r.DB.WithContext(ctxS).Where(&model.Session{Model: model.Model{ID: sessionID}}).Take(&session)
 	if err := res.Error; err != nil {
 		return e.Wrapf(err, "error getting session(id=%d) in create session events", sessionID)
 	}
 	loadSessionSpan.Finish()
 
-	clickhouseSpan, ctxW := util.StartSpanFromContext(ctxT, "public-graph.CreateSessionEvents.flush.sessionEvents")
-	clickhouseSpan.SetAttribute("NumSessionEventRows", len(events))
-	defer clickhouseSpan.Finish()
-
-	sessionEvents := []*clickhouse.SessionEventRow{}
+	var messages []kafkaqueue.RetryableMessage
 	for _, event := range events {
-		sessionEvents = append(sessionEvents, &clickhouse.SessionEventRow{
+		sessionEvent := &clickhouse.SessionEventRow{
 			ProjectID:        uint32(session.ProjectID),
 			SessionID:        uint64(session.ID),
 			SessionCreatedAt: session.CreatedAt.UnixMicro(),
 			Timestamp:        event.Timestamp,
 			Event:            event.Event,
 			Attributes:       event.Attributes,
+		}
+
+		messages = append(messages, &kafkaqueue.SessionEventRowMessage{
+			Type:            kafkaqueue.PushSessionEvents,
+			SessionEventRow: sessionEvent,
 		})
 	}
 
-	err := r.Clickhouse.WriteSessionEventRows(ctxW, sessionEvents)
+	err := r.DataSyncQueue.Submit(ctx, "", messages...)
```

Comment:
just want to think through this, but having a "" key means that messages for the same session may be inserted in order. shouldn't be a problem though, since the events are ordered by timestamp

Reply:
Sounds good! I originally had the secure session id here, but removed it because I thought it wouldn't need to be partitioned in any specific order?